### ISSUE TITLE
Pylance false positive reportUnnecessaryComparison

### ISSUE THREAD

**Issue by Questionsdenombres** [microsoft/pyright#5717]
## Environment data
Windows-10-10.0.22621-SP0
Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)])
VScode 1.81.1
Pylance v2023.8.20

## Code Snippet
```python
from typing import Literal, TypeAlias

Nature: TypeAlias = Literal["a", "b", "c"]
Result: TypeAlias = tuple[Nature, int]

def my_func(lines: list[str]) -> Result:
    result: Result = ("c", 0)

    for line in lines:
        match result[0]:
            case "b":  # reportUnnecessaryComparison
                if line.startswith("x"):
                    result = ("a", 1)
                continue
            case "c":
                if line.startswith("y"):
                    result = ("b", 2)
                continue
            case _:  # reportUnnecessaryComparison
                pass
    return result
```

## Repro Steps
Use the code snippet.
Code has no relevant function: It was reduced to the smallest piece producing the unexpected reports.
Pylance probably infers that the initial value of the `result`variable can't be updated, which is wrong.

## Expected behavior
No report.

## Actual behavior
Two reportUnnecessaryComparison

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
